include/kernel.h

```diff
@@ -586,7 +586,9 @@ struct k_thread {
 };
 
 typedef struct k_thread _thread_t;
+#ifndef CONFIG_X86
```

Comment:
This should be removed from x86 then, not here.  Surely x86 doesn't actually need a separate k_tid_t type, it's just an opaque identifier.

Reply:
Yeah, tcs definitely looks like a vestigial thing.  I guess there's some value in defining k_tid_t for the public API as an opaque handle distinct from a structure pointer (because those values are used in situations like USERSPACE where the struct itself isn't addressible), but it should be consistent and defined in only one place.  Right now we mix & match the schemes pretty badly.

Reply:
yeah, right now I just removed the second definition from x86/arch, but i needed to change k_float_enable/disable signature to use k_thread because it was really complicated to fix this circular dependency. The other option is move these functions to a better place and create dummy implementation for other platforms.

This fix is less intrusive, but we can address this problem in a further commit.